source/common/profiler/profiler.h

```diff
@@ -2,6 +2,10 @@
 
 #include <string>
 
+#if defined(TCMALLOC) && !defined(ENVOY_MEMORY_DEBUG_ENABLED)
```

Comment:
Can you add a comment explaining this? A bit counter-intuitive from the defines..